front/src/utils/trainId.ts

```diff
@@ -51,6 +51,31 @@ export const getExceptionType = (occurrence: Occurrence): 'added' | 'modified' |
 
 export const isException = (occurrence: Occurrence) => !!getExceptionType(occurrence);
 
+/**
+ * Checks if an exception is related to the path or simulation based on its exceptionChangeGroups fields.
+ * @returns boolean - `true` if the exception is related to path or simulation.
+ */
+export const isExceptionFromPathOrSimulation = (occurrence: Occurrence): boolean => {
+  const changeGroups = occurrence.exceptionChangeGroups;
+  if (!changeGroups) {
+    return false;
+  }
+
+  if (
+    changeGroups.path_and_schedule ||
+    changeGroups.options ||
+    changeGroups.constraint_distribution ||
+    changeGroups.speed_limit_tag ||
+    changeGroups.initial_speed ||
+    changeGroups.rolling_stock_category ||
+    changeGroups.rolling_stock
+  ) {
+    return true;
+  }
+
+  return false;
+};
```

Comment:
Personal preference but you could simplify the function like that
```suggestion
/**
 * Checks if an exception is related to the path or simulation.
 */
export const isExceptionFromPathOrSimulation = ({ exceptionChangeGroups }: Occurrence) =>
  exceptionChangeGroups &&
  (exceptionChangeGroups.path_and_schedule ||
    exceptionChangeGroups.options ||
    exceptionChangeGroups.constraint_distribution ||
    exceptionChangeGroups.speed_limit_tag ||
    exceptionChangeGroups.initial_speed ||
    exceptionChangeGroups.rolling_stock_category ||
    exceptionChangeGroups.rolling_stock);
```

Reply:
Done